c++/src/Statistics.hh

```diff
@@ -106,15 +106,50 @@ namespace orc {
 
     void setMinimum(T min) { _minimum = min; }
 
-    // GET / SET valueCount_
+    // GET / SET _valueCount
     uint64_t getNumberOfValues() const { return _valueCount; }
 
     void setNumberOfValues(uint64_t numValues) { _valueCount = numValues; }
 
-    // GET / SET hasNullValue_
+    // GET / SET _hasNullValue
     bool hasNull() const { return _hasNull; }
 
     void setHasNull(bool hasNull) { _hasNull = hasNull; }
+
+    void reset() {
+      _hasNull = false;
+      _hasMinimum = false;
+      _hasMaximum = false;
+      _hasSum = false;
+      _hasTotalLength = false;
+      _totalLength = 0;
+      _valueCount = 0;
+    }
+
+    // sum is not merged here as we need to check overflow
+    void merge(const InternalStatisticsImpl& other) {
+      _hasNull = _hasNull || other._hasNull;
+      _valueCount += other._valueCount;
+
+      if (other._hasMinimum) {
+        if (!_hasMinimum) {
+          _hasMinimum = _hasMaximum = true;
+          _minimum = other._minimum;
+          _maximum = other._maximum;
+        } else {
+          // all template types should support operator<
+          if (_maximum < other._maximum) {
```

Comment:
If we declare the template function in the Common.hh and define it in Common.cc. Then we have to declare it for every type as the template function is instantiated in Common.o while used in Statistics.o.